### PR TITLE
refactor: use readLocalFile instead of platform.getFile

### PR DESCRIPTION
**Pull request by rarkins**

### CODE REVIEW

File: lib/manager/cocoapods/artifacts.ts [hunk 1]
```diff
@@ -106,7 +106,7 @@ export async function updateArtifacts({
 
   const podsDir = join(dirname(packageFileName), 'Pods');
   const podsManifestFileName = join(podsDir, 'Manifest.lock');
-  if (await platform.getFile(podsManifestFileName)) {
+  if (await readLocalFile(podsManifestFileName, 'utf8')) {
```
Comment (viceice) [inline lib/manager/cocoapods/artifacts.ts:109]:
```suggestion
  if (await readLocalFile(podsManifestFileName)) {
```

as we do not need result

Comment (viceice) [inline lib/manager/cocoapods/artifacts.ts:109]: Does this always return or throw?

File: lib/manager/composer/artifacts.ts [hunk 2]
```diff
@@ -137,7 +135,7 @@ export async function updateArtifacts({
       {
         file: {
           name: lockFileName,
-          contents: await fs.readFile(localLockFileName, 'utf8'),
+          contents: await readLocalFile(lockFileName, 'utf8'),
```
Comment (viceice) [inline lib/manager/composer/artifacts.ts:138]:
```suggestion
          contents: await readLocalFile(lockFileName),
```

I think we don't need it to be string

File: lib/manager/gomod/artifacts.ts [hunk 3]
```diff
@@ -96,7 +96,7 @@ export async function updateArtifacts({
     const vendorDir = join(dirname(goModFileName), 'vendor/');
     const vendorModulesFileName = join(vendorDir, 'modules.txt');
     // istanbul ignore if
-    if (await platform.getFile(vendorModulesFileName)) {
+    if (await readLocalFile(vendorModulesFileName)) {
```
Comment (viceice) [inline lib/manager/gomod/artifacts.ts:99]: I think we should add an `existsLocalFile`function for those checks

File: lib/manager/mix/artifacts.ts [hunk 4]
```diff
@@ -41,7 +41,7 @@ export async function updateArtifacts({
     ];
   }
 
-  const existingLockFileContent = await platform.getFile(lockFileName);
+  const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
```
Comment (viceice) [inline lib/manager/mix/artifacts.ts:44]: Simple existance check?

File: lib/manager/cocoapods/artifacts.ts (hunk 1 above)
Comment (rarkins) [reply lib/manager/cocoapods/artifacts.ts:109]: It returns `null` if not found

File: lib/manager/gomod/artifacts.ts (hunk 3 above)
Comment (rarkins) [reply lib/manager/gomod/artifacts.ts:99]: Yes, I started this but it got even more complicated. I would prefer to optimize it later

File: lib/manager/cocoapods/artifacts.ts (hunk 1 above)
Comment (viceice) [reply lib/manager/cocoapods/artifacts.ts:109]: So i would suggest the following, cause we don't need extra `string` parsing
```suggestion
  if (await readLocalFile(podsManifestFileName)) {
```

**Review by viceice — APPROVED** [review]

**Pull request merged**